halyard-config/src/test/groovy/com/netflix/spinnaker/halyard/config/model/v1/ProvidersSpec.groovy

```diff
@@ -17,48 +17,47 @@
 package com.netflix.spinnaker.halyard.config.model.v1
 
 import com.netflix.spinnaker.halyard.config.model.v1.node.Providers
+import com.netflix.spinnaker.halyard.config.model.v1.providers.appengine.AppengineProvider
+import com.netflix.spinnaker.halyard.config.model.v1.providers.aws.AwsProvider
 import com.netflix.spinnaker.halyard.config.model.v1.providers.azure.AzureProvider
+import com.netflix.spinnaker.halyard.config.model.v1.providers.dcos.DCOSProvider
 import com.netflix.spinnaker.halyard.config.model.v1.providers.dockerRegistry.DockerRegistryProvider
 import com.netflix.spinnaker.halyard.config.model.v1.providers.google.GoogleProvider
 import com.netflix.spinnaker.halyard.config.model.v1.providers.kubernetes.KubernetesProvider
+import com.netflix.spinnaker.halyard.config.model.v1.providers.openstack.OpenstackProvider
+import com.netflix.spinnaker.halyard.config.model.v1.providers.oraclebmcs.OracleBMCSProvider
 import spock.lang.Specification
+import spock.lang.Unroll
 
 class ProvidersSpec extends Specification {
-  void "providers correctly reports configurable providers"() {
-    setup:
-    def providers = new Providers()
-    def iterator = providers.getChildren()
-    def kubernetes = false
-    def dockerRegistry = false
-    def google = false
-    def azure = false
-
-    when:
-    def child = iterator.getNext()
-    while (child != null) {
-      if (child instanceof KubernetesProvider) {
-        kubernetes = true
-      }
 
-      if (child instanceof DockerRegistryProvider) {
-        dockerRegistry = true
-      }
+  @Unroll()
+  void "children includes #provider.simpleName"() {
 
-      if (child instanceof GoogleProvider) {
-        google = true
-      }
-
-      if (child instanceof AzureProvider) {
-        azure = true
-      }
+    setup:
+    def iterator = new Providers().getChildren()
 
-      child = iterator.getNext()
+    when:
+    List actualProviders = []
+    def child
+    while (child = iterator.next) {
+      actualProviders << child.class
     }
```

Comment:
Ok. I might poke around and see how clean it is to make the switch. The other thing is that the name getChildren implies returning a collection rather than an iterator. I'm also not sure how to groovily use the collection operators (collect, each, etc) on just an iterator. Sounds like an interesting space to explore during pockets of free time.